src/Cache/Cache.php

```diff
@@ -143,22 +111,72 @@ public function expired(string $key): bool
 	}
 
 	/**
-	 * Checks when the cache has been created;
-	 * returns the creation timestamp on success
-	 * and false if the item does not exist
+	 * Flushes the entire cache and returns
+	 * whether the operation was successful;
+	 * this needs to be defined by the driver
 	 */
-	public function created(string $key): int|false
+	abstract public function flush(): bool;
+
+	/**
+	 * Gets an item from the cache
+	 *
+	 * <code>
+	 *   // get an item from the cache driver
+	 *   $value = $cache->get('value');
+	 *
+	 *   // return a default value if the requested item isn't cached
+	 *   $value = $cache->get('value', 'default value');
+	 * </code>
+	 */
+	public function get(string $key, mixed $default = null): mixed
 	{
-		// get the Value object
+		// get the Value
 		$value = $this->retrieve($key);
 
-		// check for a valid Value object
+		// check for a valid cache value
 		if ($value instanceof Value === false) {
-			return false;
+			return $default;
 		}
 
-		// return the expires timestamp
-		return $value->created();
+		// remove the item if it is expired
+		if ($value->expires() > 0 && time() >= $value->expires()) {
+			$this->remove($key);
+			return $default;
+		}
+
+		// return the pure value
+		return $value->value();
+	}
+
+	/**
+	 * Returns a value by either getting it from the cache
+	 * or via the callback function which then it stored in
+	 * the cache for future retrieval
+	 */
+	public function getOrSet(
+		string $key,
+		Closure $getter,
+		int $minutes = 0
+	): mixed {
+		if ($value = $this->get($key)) {
+			return $value;
+		}
```

Comment:
Can the cache value be `false`? Technically it seems possible, but if there is a real use, we need to change this line.